core/io/json.cpp

```diff
@@ -86,35 +90,47 @@ String JSON::_stringify(const Variant &p_var, const String &p_indent, int p_cur_
 		case Variant::PACKED_STRING_ARRAY:
 		case Variant::ARRAY: {
 			Array a = p_var;
+			if (p_markers.has(a.id())) {
+				r_result += "\"[...]\"";
+				ERR_FAIL_MSG("Converting circular structure to JSON.");
+			}
+
 			if (a.is_empty()) {
-				return "[]";
+				r_result += "[]";
+				return;
 			}
-			String s = "[";
-			s += end_statement;
 
-			ERR_FAIL_COND_V_MSG(p_markers.has(a.id()), "\"[...]\"", "Converting circular structure to JSON.");
+			r_result += "[";
```

Comment:
```suggestion
			r_result += '[';
```
There are similar cases to this

Reply:
Updated these to single quotes when appending characters